### PR TITLE
Feat/surface view

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: tests/mesh/test-polygonal-surface-view.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    builder.add_viewed_vertex( 1 );
+    builder.add_viewed_vertex( 2 );
+    builder.add_viewed_vertex( 3 );
+    builder.add_viewed_vertex( 4 );
+    builder.add_viewed_vertex( 5 );
+    builder.add_viewed_vertex( 6 );
+    OPENGEODE_EXCEPTION( surface.nb_vertices() == 6,
+        "[Test] PolygonalSurfaceView should have 6 vertices" );
+    const geode::Point3D answer{ { 8.1, 1.4, 4.7 } };
+    OPENGEODE_EXCEPTION( surface.point( 2 ) == answer,
```
Comment (panquez) [inline tests/mesh/test-polygonal-surface-view.cpp:65]: add test for viewed_vertex / vertex_in_view ?

File: src/geode/mesh/builder/polygonal_surface_view_builder.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        polygonal_surface_view_ =
+            &dynamic_cast< PolygonalSurfaceView< dimension >& >( mesh );
+    }
+
+    template < index_t dimension >
+    void PolygonalSurfaceViewBuilder< dimension >::do_set_point(
+        index_t /*unused*/, const Point< dimension >& /*unused*/ )
+    {
+        // polygonal_surface_view_->set_vertex( vertex_id, point, {} );
```
Comment (panquez) [inline src/geode/mesh/builder/polygonal_surface_view_builder.cpp:42]: to remove? or to prepare the future?

File: include/geode/mesh/builder/triangulated_surface_view_builder.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+namespace geode
+{
+    FORWARD_DECLARATION_DIMENSION_CLASS( Point );
+    FORWARD_DECLARATION_DIMENSION_CLASS( TriangulatedSurfaceView );
+} // namespace geode
+
+namespace geode
+{
+    /*!
+     * Implementation class for TriangulatedSurfaceBuilder using OpenGeode data
```
Comment (panquez) [inline include/geode/mesh/builder/triangulated_surface_view_builder.h:40]:
```suggestion
     * Implementation class for TriangulatedSurfaceViewBuilder using OpenGeode data

```

File: src/geode/mesh/builder/polygonal_surface_view_builder.cpp (hunk 2 above)
Comment (BotellaA) [reply src/geode/mesh/builder/polygonal_surface_view_builder.cpp:42]: API is requested to satify inheritance. Implementation will be added later.

**Pull request merged**